tests/monit/test_monit_service_status.py

```diff
@@ -17,10 +17,10 @@ def test_monit_service_status(duthost):
     @summary: Test the running status of Monit service by analyzing the command
               output of "sudo systemctl status monit.service | grep Active".
     """
-    monit_service_status_info = duthost.shell("sudo systemctl status monit.service | grep Active")
+    monit_service_status_info = duthost.shell("sudo monit status", module_ignore_errors=True)
 
-    status_line = monit_service_status_info["stdout_lines"][0].strip()
-    if "active" in status_line:
+    exit_code = monit_service_status_info["rc"]
+    if exit_code == 0:
         logger.info("Monit service is running.")
     else:
-        pytest.fail("Monit service is not running: '{}'".format(status_line))
+        pytest.fail("Monit service is not running.")
```

Comment:
A return code of 0 means Monit is running and healthy. A non-zero return code does not necessarily mean Monit is not running. It could be running with a bad config.

Reply:
i think better to use the wrapper. pytest_fail?

Reply:
There is no `pytest_fail` wrapper. I believe you're thinking of the `pytest_assert` wrapper.

Reply:
I don't think we gain much from logging that Monit is running. I think this if/else should be replaced with:

```python
pytest_assert(exit_code == 0, "Monit is either not running or not configured correctly")
```

Reply:
> A return code of 0 from `sudo monit status` means Monit is running and healthy. A non-zero return code does not necessarily mean Monit is not running. It could be running with a bad config.

@jleveque You mentioned that if the return code is non-zero, Monit could be running with a bad config. "bad config" at here means someone modified the Monit configuration files and introduced syntax errors or other errors. But they did not run the command `sudo systemctl restart monit.service`. Although Monit is still in the running status, return code of the command `sudo monit status` will be non-zero, right?

Reply:
If Monit is running and the currently loaded Monit configuration contains syntax errors or similar, `sudo monit status will return a non-zero value.

Reply:
That's what I thought and tested.